xmrstak/misc/environment.hpp

```diff
@@ -12,35 +12,30 @@ class params;
 
 struct environment
 {
-
-	static environment& inst()
-	{
-		static environment env;
```

Comment:
Normally this is the comon way to create a singleton and is free of memory leeks. Is there a reason why you have changed it?